src/credentials/FabricTable.cpp

```diff
@@ -212,26 +276,30 @@ CHIP_ERROR FabricInfo::GeneratePeerId(FabricId fabricId, NodeId nodeId, PeerId *
 
 CHIP_ERROR FabricInfo::DeleteFromStorage(PersistentStorageDelegate * storage, FabricIndex fabricIndex)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    DefaultStorageKeyAllocator keyAlloc;
+
+    // Try to delete all the state even if one of the deletes fails.
+    typedef const char * (DefaultStorageKeyAllocator::*KeyGetter)(FabricIndex);
+    constexpr KeyGetter keyGetters[] = { &DefaultStorageKeyAllocator::FabricNOC, &DefaultStorageKeyAllocator::FabricICAC,
```

Comment:
Can this be something that lives inside the key allocator? Might need some more structuring in there.

Reply:
https://github.com/project-chip/connectedhomeip/issues/16078 filed to track.